tests/test_json_validation.py

```diff
@@ -177,20 +177,59 @@ def test_resolve_refs():
     one = do_resolve("1.json")
     assert one["properties"] == {"1_prop": {"2_prop": {"3_prop": {"type": "string"}}}}
 
+def test_get_values_for_path_pattern():
+    v = _Validator({})
+    # Absolute path, all strings
+    assert v._get_values_for_path_pattern("/a/b/c", {"a": {"b": {"c": "foo", "d": "bar"}}}) == set([repr("foo")])
+    # Absolute path, array index
+    assert v._get_values_for_path_pattern("/a/1/b", {"a": [{"b": "foo"}, {"b": "bar"}]}) == set([repr("bar")])
```

Comment:
Seems this will break
```suggestion
    assert v._get_values_for_path_pattern("/a/1/b", {"a": [{"b": "foo"}, {"b": "bar"}]}) == set([repr("bar")])
    # dict with a key that looks like integer
    assert v._get_values_for_path_pattern("/a/0/1", {"a": [{"1": "foo"}, {"1": "bar"}]}) == set(repr("foo")])
```
Though I'm not sure how json-pointer spec suggests this should be handled

Reply:
Actually, this test passes. The key check first assumes any non-`*` value is a string, then treats it as an index.